content/en/blog/docker-multi-arch/index.md

```diff
@@ -0,0 +1,114 @@
+
+{{< imgproc arch-overview Fit "625x625" >}}
+<em>High-level architecture of Findy Agency. Setting up agency to localhost is most straightforward with the help of container orchestration tool.</em>
+{{< /imgproc >}}
+
+Until recently, we were happy with our image-building pipeline. Local environments booted up with ease, and the deployment pipeline rolled out updates beautifully. Then one day, our colleague with an M1-equipped Mac tried out the docker-compose script. Running the agency images in an Arm-based architecture was something we hadn't considered. We built our Docker images for amd64 architecture, while M1 Macs expect container images for arm64 CPU architecture. It became clear we needed to support also the arm64, as we knew that the popularity of the M1 chipped computers would only increase in the future.
+
+## Multi-architecture Support in Docker
+
+Typically, when building images for Docker, the image inherits the architecture type from the building machine. And as each processor architecture requires a dedicated Docker image, one needs to build a different container image for each target architecture. To avoid the hassle with the multiple images, Docker has added support for multi-architecture images. It means that there is a single image in the registry, but it can have many variants. Docker will automatically choose the appropriate architecture for the processor and platform in question and pull the correct variant.
+
+Ok, so Docker takes care of the image selection when running images. How about building them then? There are [three strategies](https://docs.docker.com/buildx/working-with-buildx/#build-multi-platform-images).
+1. **QEMU emulation support in the kernel**: QEMU works by emulating all instructions of a foreign CPU instruction set on the host processor. For example, it can emulate ARM CPU instructions on an x86 host machine. The QEMU emulator enables building images that target another architecture than the host.
+1. **Multiple native nodes using the same builder instance**: With this approach, multiple hosts that run on different CPU architectures execute the build. The build time is faster. The drawback is that it requires access to native nodes.
+1. **Stage in a Dockerfile for cross-compilation**: This option is possible with languages that support cross-compilation. Arguments exposing the build and the target platforms are automatically available to the build stage. The build command can utilize these parameters to build the binary for the correct target.
+
+From these three options, we chose the first one, as it seemed the most straightforward route. However, in our case, the third option might have worked as well since we are building with tools that support cross-compilation, Rust, and GoLang. 
```

Comment:
At this point of the text, I would like to see some sort of pros and cons for each option. Excellent stuff! I could guess that 1 is slower than third, but what else, etc.